lm_eval/models/huggingface.py

```diff
@@ -678,39 +678,61 @@ def _create_tokenizer(
             )
         return None
 
-    def _detect_batch_size(self, requests=None, pos: int = 0):
-        if requests:
+    def _detect_batch_size(self, requests=None, pos: int = 0) -> int:
+        if len(requests[0]) == 3:  # logprob evals
```

Comment:
`requests`by default is None, and is not indexable

Reply:
By the way, how this detection will work if two tasks passed as `--tasks` option, one with logprob evals, and the other one with generative evals? Will tasks ordering affect this detection?

Reply:
Hey ! Thanks for the catch. I proposed a fix in the latest commit.

> By the way, how this detection will work if two tasks passed as --tasks option, one with logprob evals, and the other one with generative evals? Will tasks ordering affect this detection?

It will compute the batch size for both. The behaviour does not change for loglikelihood tasks. However, the batch size is recomputed for each generate until tasks.